src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCopy = { ...state };
+  const stateArray = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        for (const key in stateCopy) {
+          delete stateCopy[key];
+        }
+        stateArray.push({ ...stateCopy });
```

Comment:
you can move push after switch. you make the same operation regardless of the action type